src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2921,6 +2926,29 @@ InfoSub::pointer NetworkOPsImp::addRpcSub (
     return rspEntry;
 }
 
+bool NetworkOPsImp::tryRemoveRpcSub (std::string const& strUrl)
+{
+    ScopedLockType sl (mSubLock);
+    auto pInfo = findRpcSub(strUrl);
+
+    // check to see if any of the stream maps still hold a weak reference to
+    // this entry before removing
+    if ( pInfo &&
+         mSubLedger.find(pInfo->getSeq()) == mSubLedger.end() &&
```

Comment:
thanks for this change - I've pulled-in the array-of-maps version and will submit shortly